image/oci.go

```diff
@@ -148,7 +154,7 @@ func (m *manifestOCI1) UpdatedImage(options types.ManifestUpdateOptions) (types.
 		if len(copy.LayersDescriptors) != len(options.LayerInfos) {
 			return nil, errors.Errorf("Error preparing updated manifest: layer count changed from %d to %d", len(copy.LayersDescriptors), len(options.LayerInfos))
 		}
-		copy.LayersDescriptors = make([]descriptor, len(options.LayerInfos))
+		copy.LayersDescriptors = make([]descriptorOCI1, len(options.LayerInfos))
 		for i, info := range options.LayerInfos {
 			copy.LayersDescriptors[i].Digest = info.Digest
 			copy.LayersDescriptors[i].Size = info.Size
```

Comment:
Actually, yes, the definition of `LayerInfos` says that `URLs` should also be copied. But that’s unrelated to this PR, so I’m fine with both keeping this as is and adding the `URLs` copy here.

(That seems to have the effect of mostly dropping the URLs field whenever we upload a blob, even if it would make sense to keep the original URLs. But let’s worry about that in some other PR or issue.)